forge-ai/src/main/java/forge/ai/ability/ChooseCardAi.java

```diff
@@ -58,11 +59,27 @@ protected boolean checkApiLogic(final Player ai, final SpellAbility sa) {
     protected boolean checkAiLogic(final Player ai, final SpellAbility sa, final String aiLogic) {
         final Card host = sa.getHostCard();
         final Game game = ai.getGame();
+
         ZoneType choiceZone = ZoneType.Battlefield;
+        CardCollectionView choices = new CardCollection();
         if (sa.hasParam("ChoiceZone")) {
-            choiceZone = ZoneType.smartValueOf(sa.getParam("ChoiceZone"));
+            String definedChoiceZone = sa.getParam("ChoiceZone");
```

Comment:
try to use classes own helpers instead of manually coding like `listValueOf` here

Reply:
Done :) Thanks for the heads up, didn't realize listValueOf was a thing :)

Reply:
Still seems way overcomplicated :p
why not just have the code look like this?
```
        List<ZoneType> choiceZone = Lists.newArrayList(ZoneType.Battlefield);
        if (sa.hasParam("ChoiceZone")) {
            choiceZone = ZoneType.listValueOf(sa.getParam("ChoiceZone"));
        }
        CardCollectionView choices = game.getCardsIn(choiceZone);
```

Reply:
Ah, indeed :) Thanks a lot for the suggestion! Implemented.